### PR TITLE
Kd additional features

### PR DESCRIPTION
**Pull request by SKholkin**
### Changes

- Ignoring model outputs with number of dimension < 2 (loss values) and some additional
- ability to call compression_ctrl.loss() before training through returning zero tensor instead of raising an error

### Reason for changes

- Fixing a bug
- Improvement of user experience

### Related tickets

68004

### Tests

test_kd_output_shapes_handling Checks ignorance behavior (kd loss is zero) for different model output shape sizes
test_kd_softmax_loss_ignores_incompatible_outputs is deleted because test_kd_output_shapes_handling generalizes it.

### CODE REVIEW

File: tests/torch/test_knowledge_distillation.py [hunk 1]
```diff
@@ -373,23 +371,48 @@ def forward(self, *args, **kwargs):
...
+class CustomOutputWeightenedModel(torch.nn.Module):
+    def __init__(self, input_shape: List[int], target_output_shapes: List[int]):
+        super().__init__()
+        self.weights = nn.ParameterDict()
+        self.linears = nn.ModuleDict()
+        self.target_output_shapes = target_output_shapes
+        self.linears['4'] = torch.nn.Linear(in_features=input_shape[3], out_features=input_shape[3])
+        self.linears['3'] = torch.nn.Linear(in_features=input_shape[3], out_features=1)
+        self.linears['2'] = torch.nn.Linear(in_features=input_shape[2], out_features=1)
+        self.linears['1'] = torch.nn.Linear(in_features=input_shape[1], out_features=1)
```
Comment (vshampor) [inline tests/torch/test_knowledge_distillation.py:383]: Please refactor the test or add explicit comments explaining just what is happening here - it's hard to read the code handling in magic constants such as 4, 3, '4' etc

**Review by vshampor — CHANGES_REQUESTED** [review]

File: tests/torch/test_knowledge_distillation.py (hunk 1 above)
Comment (SKholkin) [reply tests/torch/test_knowledge_distillation.py]: Test has been changed

**Review by vshampor — APPROVED** [review]

**Review by alexsu52 — APPROVED** [review]

**Comment by SKholkin** [comment]
Jenkins please retry a build

File: nncf/torch/knowledge_distillation/knowledge_distillation_loss.py [hunk 2]
```diff
@@ -46,6 +46,11 @@ def kd_loss_fn(ref_output: torch.Tensor, compressed_model_output: torch.Tensor):
         else:
             def kd_loss_fn(ref_output: torch.Tensor, compressed_model_output: torch.Tensor):
                 mse = torch.nn.MSELoss()
+                if len(ref_output.shape) < 2:
+                    nncf_logger.debug("Incompatible shape (compressed - {}, ref - {}) of the model output tensor "
```
[the remark this answers is not in the document]
Comment (ljaljushkin) [reply nncf/torch/knowledge_distillation/knowledge_distillation_loss.py:50]: It would be more clear, if error message said about number of dimensions. Also it might be not clear for the user what is `ref`. Probably there should be `student` and `teacher`

**Review by ljaljushkin — CHANGES_REQUESTED** [review]

File: nncf/torch/knowledge_distillation/knowledge_distillation_loss.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (SKholkin) [reply nncf/torch/knowledge_distillation/knowledge_distillation_loss.py:50]: Making this message a warning instead of debug info will make it appear every train step and console logging will not be readable. So I suggest to let it stay as a debug info.

WARNING:nncf:Incompatible number of dimensions of the model output tensor for softmax KD (student - torch.Size([]), teacher - torch.Size([]), number of dims 0 should be == 2) - ignoring!
WARNING:nncf:Incompatible number of dimensions of the model output tensor for softmax KD (student - torch.Size([]), teacher - torch.Size([]), number of dims 0 should be == 2) - ignoring!
  9%|██████████████▍                                                                                                                                            | 37/396 [00:51<08:11,  1.37s/it, loss=0.221]
WARNING:nncf:Incompatible number of dimensions of the model output tensor for softmax KD (student - torch.Size([]), teacher - torch.Size([]), number of dims 0 should be == 2) - ignoring!
WARNING:nncf:Incompatible number of dimensions of the model output tensor for softmax KD (student - torch.Size([]), teacher - torch.Size([]), number of dims 0 should be == 2) - ignoring!

**Review by ljaljushkin — APPROVED** [review]

**Comment by SKholkin** [comment]
Jenkins please retry a build

**Comment by SKholkin** [comment]
Jenkins please retry a build
